### ISSUE TITLE
20190314 - meeting

### ISSUE THREAD

**Issue by dpaez** [dat-land/comm-comm#38]
Just an hour hanging out and talking about DAT stuff.

Time: 14 Mar 2019, 18:00 GMT

https://meet.jit.si/DATcomm-comm

| Timezone | Time | 
|---|---|
| Auckland | 07:00 (am) - Mar. 15 _(Fri)_ |
| Tokyo | 03:00 (am) |
| Hong Kong | 02:00 (am) |
| Karachi | 23:00 (pm) - Mar. 14 _(Thu)_ |
| Dubai | 22:00 (pm) |
| Moscow, Nairobi | 21:00 (pm) |
| Berlin | 19:00 (pm) |
| London | 18:00 (pm) |
| Sao Paulo | 15:00 (pm) |
| New York | 14:00 (pm) |
| Mexico City | 12:00 (pm) |
| Los Angeles | 11:00 (am) |
| Honolulu | 08:00 (am) |

**Comment by perguth** [comment]
Use this link to start out muted: https://meet.jit.si/DATcomm-comm#config.startWithAudioMuted=true&config.startWithVideoMuted=true

**Comment by RangerMauve** [comment]
So, some of the stuff that I think would be useful for different companies / groups / independents to collaborate on would be:

- A standard set of tools for getting Dat running in browsers
  - Discovery
  - Efficient storage
- A standard for pinning hypercores  / hyperdrives
  - Make make pinning service that different projects are comfortable with using
  - Make it easy to start the service as a daemon in the dat-cli
- Docs!
  - As a new user to Dat, there should be a page that has a high level description of the parts they could be using
  - A quickstart for making a web application with Dat
- Other environments
  - Helping move [the rust implementation](https://github.com/datrs) forward
  - An easy to use library for using Dat in React-Native

**Comment by RangerMauve** [comment]
The stuff in the [planning](https://github.com/datproject/planning/) repo should also be on the list. :P

**Comment by karissa** [comment]
It would also be nice to have a threat model for the dat protocol and have a diagram people can see, as well as a user guide for increasing privacy and the resulting trade-offs in peer discovery.

**Comment by martinheidegger** [comment]
@RangerMauve  "A standard for pinning hypercores / hyperdrives" Can you elaborate on that? It is not clear what "pinning" means?! Maybe add here: https://github.com/datprotocol/DEPs/pull/5

**Comment by martinheidegger** [comment]
@yoshuawuyts can you help us out about the rust implementation? What is the state? How can people contribute?

**Comment by martinheidegger** [comment]
I just chatted with @yoshuawuyts : the rust implementation is waiting on https://areweasyncyet.rs/ before the networking implementation can be done and hypercore tests can be worked on. You can ping him in the chat if you want to work/help on the rust implementation.

**Comment by dpaez** [comment]
Hi all, we were talking with @martinheidegger on what things we can do to solve this issue. This is, to work mostly with the items mentioned in @RangerMauve's [comment](https://github.com/dat-land/comm-comm/issues/38#issuecomment-473017953).

So, according to that we have the following _proposed_ actions:
> A standard set of tools for getting Dat running in browsers
Discovery
Efficient storage

We can write a post about all the recent progress and efforts we are doing with discovery, things like:
-  dat-js
- discovery-swarm-web
- discovery-swarm-webrtc
About `storage`, there is a new module: https://github.com/random-access-storage/random-access-web. 

We can also use an example app, ideally we should use some simple app using the new dat-js stuff.

> A standard for pinning hypercores / hyperdrives
Make make pinning service that different projects are comfortable with using
Make it easy to start the service as a daemon in the dat-cli

We need more info regarding this. In order to choose the best way to help.

> Docs!
As a new user to Dat, there should be a page that has a high level description of the parts they could be using
A quickstart for making a web application with Dat

We can open an issue on the dat site maybe where we can focus on the quickstart part, maybe like a tree with links, so if the reader wants to create a CLI, can jump to that specific part, or if they want to create something on the web. Besides that, it would be really nice to have some designer involved on this task as a way to help us create really good and _easy-to-digest_ high level descriptions (as images maybe). 

> Other environments
Helping move the rust implementation forward
An easy to use library for using Dat in React-Native

Finally, again, we should link (somewhere in the dat blog) to the resource that @martinheidegger mention before (https://areweasyncyet.rs/). After that, it would be nice to maintain a good communication with others developers involved in this _bridge_. Maybe @yoshuawuyts can join any comm-comm meeting and share some news :). About the rn part, I think @RangerMauve wants to work on a new post explaining his adventure with Dat & rn. So a cool reading is on the way!

That's the plan, feel free to share your ideas! 👍

**Comment by RangerMauve** [comment]
Regarding the pinning service, I want to work off of what @pfrazee started with [DEP 0003](https://github.com/datprotocol/DEPs/blob/master/proposals/0003-http-pinning-service-api.md) and integrate something into the CLI. So far I'm thinking of using a super basic service for local use that doesn't have a UI or anything and can be run as a daemon from the dat-cli. I want to integrate the discovery server from `discovery-swarm-web` as part of the CLI as well.

Regarding the react-native stuff, my experiment with Datmobille is kind of on hold because the performance wasn't good enough. @allain has been looking into improving the performance of `random-access-rn-file` which might make it more bareable. I'll also need to revisit the discovery mechanism to see if I can move more stuff into react-native modules.

**Comment by dpaez** [comment]
> Hi all, we were talking with @martinheidegger on what things we can do to solve this issue. This is, to work mostly with the items mentioned in @RangerMauve's [comment](https://github.com/dat-land/comm-comm/issues/38#issuecomment-473017953).
> 
> So, according to that we have the following _proposed_ actions:
> 
> > A standard set of tools for getting Dat running in browsers
> > Discovery
> > Efficient storage
> 
> We can write a post about all the recent progress and efforts we are doing with discovery, things like:
> 
> * dat-js
> * discovery-swarm-web
> * discovery-swarm-webrtc
>   About `storage`, there is a new module: https://github.com/random-access-storage/random-access-web.

[**UPDATE**]: @RangerMauve latest post on dat blog covers all of these (^) points really well!

https://blog.datproject.org/2019/04/14/getting-dat-to-work-in-new-environments/

**Comment by martinheidegger** [comment]
Closing this issue. Let me know if this is too early.
